### PR TITLE
MSPSDS-976 prevent view confidential files

### PR DESCRIPTION
**Pull request by MatthewBurstein**
## Checklist:

- [X] Automated checks are passing locally.
- [X] CHANGELOG updated if change is worth telling users about.
### General testing
- [X] Test without javascript
- [X] Test on small screen
### Accessibility testing
- [X] Works keyboard only
- [X] Tested with one screen reader
- [X] Zoom page to 400% - content still visible
- [ ] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

### CODE REVIEW

File: mspsds-web/app/views/documents/_document_card.html.slim [hunk 1]
```diff
@@ -0,0 +1,23 @@
+- sensitive = document.metadata[:has_consumer_info]
```
Comment (kicferk1) [inline mspsds-web/app/views/documents/_document_card.html.slim]: Can probably use `ruby:` block here, and move `document.metadata["safe"]` here as well.

File: mspsds-web/app/views/documents/_document_card.html.slim [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  .govuk-grid-column-one-third
+    = render "documents/document_preview", document: document, dimensions: "300x200",
+            sensitive: sensitive
+  .govuk-grid-column-two-thirds
+    h2.govuk-heading-m[class="govuk-!-margin-bottom-1"] = title
+    span.govuk-hint[class=" govuk-!-font-size-16"] = formatted_file_updated_date(document)
+    - if sensitive
+      p = render "components/govuk_badge", text: "RESTRICTED ACCESS", classes: "govuk-badge--grey"
+    p = description
+    - unless sensitive
```
Comment (kicferk1) [inline mspsds-web/app/views/documents/_document_card.html.slim]: else instead?

File: mspsds-web/app/views/investigations/activities/_activity_card.html.slim [hunk 3]
```diff
@@ -25,5 +26,12 @@ li
       - if activity.business_id
         = link_to "View business details", business_url(activity.business_id), class: "mspsds-block-link"
       - if activity.has_attachment?
-        - activity.attachments.each do |name, attachment|
-          = link_to "View #{name}", attachment, class: "mspsds-block-link", target: "_blank", rel: 'noopener'
+        - if activity.sensitive?
+          - attachments_string = pluralize(activity.attachments.count, 'attachment')
+          br
+          p
```
Comment (kicferk1) [inline mspsds-web/app/views/investigations/activities/_activity_card.html.slim]: Do we need br and p? On a tangent, this file is quite messy, is it time for it to be refactored into something more readable?

File: mspsds-web/app/views/investigations/activities/_activity_card.html.slim [hunk 4]
```diff
@@ -13,8 +13,9 @@ li
         = markdown activity.body if activity.body
   - else
     p
-      - if activity.sensitive_body?
-        | Consumer contact details hidden to comply with GDPR legislation.
+      - if activity.sensitive?
+        = render "components/govuk_badge", text: "RESTRICTED ACCESS", classes: "govuk-badge--grey"
+        p Consumer contact details hidden to comply with GDPR legislation.
```
Comment (broder) [inline mspsds-web/app/views/investigations/activities/_activity_card.html.slim]: Does this result in `p` tags within `p` tags?

File: mspsds-web/test/system/record_email_correspondence_test.rb [hunk 5]
```diff
@@ -122,6 +122,20 @@ class RecordEmailCorrespondenceTest < ApplicationSystemTestCase
     assert test_request.save
   end
 
+  test "conceals information on phonecalls with customer info" do
```
Comment (broder) [inline mspsds-web/test/system/record_email_correspondence_test.rb]: "emails with customer info"?

File: mspsds-web/app/views/investigations/activities/_activity_card.html.slim [hunk 6]
```diff
@@ -13,8 +13,9 @@ li
         = markdown activity.body if activity.body
   - else
     p
-      - if activity.sensitive_body?
-        | Consumer contact details hidden to comply with GDPR legislation.
+      - if activity.sensitive?
+        = render "components/govuk_badge", text: "RESTRICTED ACCESS", classes: "govuk-badge--grey"
```
Comment (broder) [inline mspsds-web/app/views/investigations/activities/_activity_card.html.slim]: The "RESTRICTED ACCESS" uppercasing will also be done by the CSS - maybe we should just have it as "Restricted access" here?

File: shared-web/app/assets/application/stylesheets/badge.scss [hunk 7]
```diff
@@ -0,0 +1,25 @@
+.govuk-badge {
```
Comment (jasiekmiko) [inline shared-web/app/assets/application/stylesheets/badge.scss]: @MatthewBurstein hey, sorry I'm only picking up on this PR now. Wandering if we took this CSS from somewhere or if it's all our own?

**Pull request merged**